e2e/test/scenarios/onboarding/search/search.cy.spec.js

```diff
@@ -242,52 +269,143 @@ describe("scenarios > search", () => {
+                cy.findByText("Apply all filters").click();
+              });
+
+              getSearchBar().clear().type("e{enter}");
+              cy.wait("@search");
+
+              cy.url().should("include", `type=${filterName}`);
+
+              cy.findAllByTestId("result-link-info-text").each(result => {
+                cy.wrap(result).should("contain.text", resultInfoText);
+              });
+
+              cy.findAllByTestId("type-sidebar-item").should("have.length", 2);
+              cy.findByTestId("type-sidebar").within(() => {
+                cy.findByText(sidebarLabel).should("exist");
+              });
             });
+          },
+        );
+      });
 
-            getSearchBar().clear().type("e{enter}");
-            cy.wait("@search");
+      describe("verified filter", () => {
+        it("should filter only for `Verified` assets", () => {
+          cy.visit("/");
 
-            cy.url().should("include", `type=${filterName}`);
+          cy.findByTestId("search-bar-filter-button").click();
+          getSearchModalContainer().within(() => {
+            cy.findByText("Only verified items").click();
+            cy.findByText("Apply all filters").click();
+          });
 
-            cy.findAllByTestId("result-link-info-text").each(result => {
-              cy.wrap(result).should("contain.text", resultInfoText);
-            });
+          getSearchBar().clear().type("Orders{enter}");
+          cy.wait("@search");
+
+          cy.url().should("include", `verified=true`);
+
+          const verifiedItem = cy.findByTestId("search-result-item");
+          verifiedItem.within(() => {
+            cy.findByLabelText("verified icon").should("exist");
+          });
+        });
+
+        it("should show verified and unverified assets when `All items` is selected", () => {
+          cy.visit("/");
 
-            cy.findAllByTestId("type-sidebar-item").should("have.length", 2);
-            cy.findByTestId("type-sidebar").within(() => {
-              cy.findByText(sidebarLabel).should("exist");
+          cy.findByTestId("search-bar-filter-button").click();
+          getSearchModalContainer().within(() => {
+            cy.findByText("All items").click();
+            cy.findByText("Apply all filters").click();
+          });
+
+          getSearchBar().clear().type("Orders{enter}");
+          cy.wait("@search");
+
+          cy.url().should("not.include", `verified=true`);
+
+          let verifiedElementCount = 0;
+          let unverifiedElementCount = 0;
+
+          cy.findAllByTestId("search-result-item")
+            .each($el => {
+              if ($el.find('[aria-label="verified icon"]').length) {
+                verifiedElementCount++;
+              } else {
+                unverifiedElementCount++;
+              }
+            })
+            .then(() => {
+              expect(verifiedElementCount).to.eq(1);
+              expect(unverifiedElementCount).to.be.gt(0);
```

Comment:
I don't see we count verified/unverified items for the test `"should filter only for `Verified` assets"`. It feels like that test only checks if the verified item is found in the search result, but there's no apparent assertion to assert that there are **no** non-verified items in the search result.

I kind of expect the test body to be similar between these 2 tests since they're almost the same but just asserting different number of non-verified items.

Reply:
I see what you mean - the test for `should filter only for "Verified" assets` checks that _all_ elements within the search results have a `Verified` checkmark instead of counting how many search results we have. If there's at least one unverified element then it'll fail.

With the other test, we need to make sure that there's at least one of each result. So, we have to count the results to have some confidence that we're not just returning verified results when we know that we want to have "unverified" results too